pkg/chunk/cassandra/storage_client.go

```diff
@@ -48,6 +62,20 @@ func (cfg *Config) session() (*gocql.Session, error) {
 	cluster.Consistency = consistency
 	cluster.BatchObserver = observer{}
 	cluster.QueryObserver = observer{}
+	cluster.DisableInitialHostLookup = cfg.disableInitialHostLookup
+
+	if cfg.ssl {
+		cluster.SslOpts = &gocql.SslOptions{
+			CaPath:                 cfg.caPath,
+			EnableHostVerification: cfg.hostVerification,
+		}
+	}
+	if cfg.auth {
+		cluster.Authenticator = gocql.PasswordAuthenticator{
+			Username: cfg.username,
+			Password: cfg.password,
+		}
+	}
```

Comment:
Would be good to extract as much of this as we can into a helper function, to reduce repetition.

Reply:
DisableInitialHostLookup still found at java datastatx driver, right ?